ndcube/utils/wcs.py

```diff
@@ -197,18 +197,25 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    extra_coords_list = [] # initiating new list
+    for slice_element in item_: # check item_ slices for dropped axes if any
+        if (slice_element.stop - slice_element.start) == 1:
+           extra_coords_list.append(slice_element)
```

Comment:
This appends the ```slice``` to the list which isn't what we want.  We want to append a tuple of ```(name, data_axis, coord_value)``` where ```name``` is the real world axis name.  This can be got from the old WCS object's CTYPE attribute.  The ```data_axis``` should always be ```None``` since the data axis is being sliced out.  And the ```coord_value``` must be derived from the old WCS's ```all_pix2world``` method.

Reply:
Yup, understood. (Actually was stuck in this part a bit before.) Attempted to follow instructions to redefine how the `extra_coords_list` is generated. Might have made a mistake or two though, so please do check.